src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfCodec.java

```diff
@@ -0,0 +1,376 @@
+            // Check the normal commented fields:
+            isValid = header[0].startsWith("##description:");
+            isValid = isValid && header[1].startsWith("##provider: GENCODE");
+
+            isValid = isValid && header[2].startsWith("##contact: gencode");
+            isValid = isValid && header[2].endsWith("@sanger.ac.uk");
+
+            isValid = isValid && header[3].startsWith("##format: gtf");
+            isValid = isValid && header[4].startsWith("##date:");
+        }
+
+        return isValid;
+    }
+
+    /**
+     * Aggregates the given feature sets into a single gene feature.
+     *
+     * The given gene is updated using modifiers.
+     * {@code exonStore} and {@code leafFeatureStore} are cleared of all data.
+     *
+     * @param gene {@link GencodeGtfGeneFeature} into which to aggregate features.
+     * @param transcript {@link GencodeGtfTranscriptFeature} to insert into {@code gene}
+     * @param exonStore {@link List} of {@link GencodeGtfExonFeature}s to insert into corresponding {@link GencodeGtfTranscriptFeature} {@code transcript}
+     * @param leafFeatureStore {@link List} of {@link GencodeGtfFeature}s to insert into corresponding {@link GencodeGtfExonFeature} objects in {@code exonStore}
+     */
+    private void aggregateRecordsIntoGeneFeature(final GencodeGtfGeneFeature gene,
+                                                                  final GencodeGtfTranscriptFeature transcript,
+                                                                  final ArrayList< GencodeGtfExonFeature > exonStore,
+                                                                  final ArrayList< GencodeGtfFeature > leafFeatureStore ) {
+
+        // OK, we go through the record and consolidate the sub parts of the record.
+        // We must consolidate these records through grouping by genomic position.
+
+        // Loop through the Exons and put the correct leaf features into each:
+        for ( GencodeGtfExonFeature exon : exonStore ) {
+            for (Iterator<GencodeGtfFeature> iterator = leafFeatureStore.iterator(); iterator.hasNext(); ) {
+
+                GencodeGtfFeature feature = iterator.next();
+
+                // Features that are within the extents of an exon belong in that exon:
+                if ( exon.contains(feature) ) {
+
+                    GencodeGtfFeature.FeatureType featureType = feature.getFeatureType();
+
+                    // Add the feature to the correct place in the exon:
+                    switch (featureType) {
+                        case cds:
+                            exon.setCds((GencodeGtfCDSFeature) feature);
+                            break;
+                        case start_codon:
+                            exon.setStartCodon((GencodeGtfStartCodonFeature) feature);
+                            break;
+                        case stop_codon:
+                            exon.setStopCodon((GencodeGtfStopCodonFeature) feature);
```

Comment:
They will not.  Yes - it should be an error, otherwise data would be overwritten / lost.